lgc/interface/lgc/Pipeline.h

```diff
@@ -562,24 +562,40 @@ class Pipeline {
   virtual void setStateFromModule(llvm::Module *module) = 0;
 
   // -----------------------------------------------------------------------------------------------------------------
-  // IR link and generate pipeline methods
+  // IR link and generate pipeline/library methods
+
+  // Mark a function as a shader entry-point. This must be done before linking shader modules into a pipeline
+  // with irLink(). This is a static method in Pipeline, as it does not need a Pipeline object, and can be used
+  // in the front-end before a shader is associated with a pipeline.
+  //
+  // @param func : Shader entry-point function
+  // @param stage : Shader stage
+  static void markShaderEntryPoint(llvm::Function *func, ShaderStage stage);
 
   // Link the individual shader modules into a single pipeline module. The front-end must have
   // finished calling Builder::Create* methods and finished building the IR. In the case that
   // there are multiple shader modules, they are all freed by this call, and the linked pipeline
   // module is returned. If there is a single shader module, this might instead just return that.
-  // Before calling this, each shader module needs to have one global function for the shader
-  // entrypoint, then all other functions with internal linkage.
-  // Returns the pipeline module, or nullptr on link failure.
   //
-  // @param modules : Array of {module, shaderStage} pairs
+  // Before calling this, each shader module needs to have exactly one global function for the shader
+  // entry-point that was marked by calling markShaderEntryPoint(). Any other functions in the module
+  // must not have a non-default DLL storage class, and typically have internal linkage. However, for
+  // a compute shader containing functions accessed by libraries, those functions need to be global.
+  //
+  // In the case of a compute library, there is no shader entry-point marked by calling
+  // markShaderEntryPoint(). All functions must have default DLL storage class, and any that need to
+  // be externally accessible need to be global.
```

Comment:
In LLVM IR, all functions are always global :) I think you want to say something about linkage here...